sklearn/linear_model/_cd_fast.pyx

```diff
@@ -695,8 +695,8 @@ def enet_coordinate_descent_gram(floating[::1] w,
 
 def enet_coordinate_descent_multi_task(
         floating[::1, :] W, floating l1_reg, floating l2_reg,
-        np.ndarray[floating, ndim=2, mode='fortran'] X,  # TODO: use views with Cython 3.0
```

Comment:
As noted in the comment, we need to wait for Cython 3.0 to do this, because of fused types and memmaps not working for Cython < 3.0: https://github.com/scikit-learn/scikit-learn/issues/10624#issuecomment-449407648

In this case, the following will fail on this PR:

```python
import numpy as np
from sklearn.linear_model import lasso_path

X = np.array([[1, 2.3], [2, 5.4], [3.1, 4.3], [5.4, 4.3]], order='F')
X.flags["WRITEABLE"] = False
y = np.array([[1.0, 2.0], [2.0,3.0], [4, 5]], order='F')

_, coef_path, _ = lasso_path(X, y, alphas=[5., 1., .5], check_input=False)
```

but passes on `main`.